cli/azd/docs/featue-toggle.md

```diff
@@ -0,0 +1,61 @@
+# Feature Toggle
+
+The Azure Developer CLI includes a `feature toggle` which allows the cli to ship `in-preview` ( _a.k.a. experimental_ ). This document provides the feature design for it.
+
+## Main requirement
+
+As a customer, I can explicitly ask `azd` to unblock and show any _experimental_ functionality. The features becoming available are not finalized, so they can be changed or removed in future releases.
+
+The strategy to ask `azd` to `toggle` between modes is:
+
+> azd config set experimental on
+
+and
+
+> azd config set experimental off
+
+## Experimental features
+
+When toggling experimental mode `on`, the Azure Developer CLI includes the next type of features:
+
+1. Command or sub-command
+    - The command becomes available to be run just like any other command.
+    - Customers are **not** expected to know when a command or sub-command is experimental by just typing the command. This means:
+      - There are not suffix or prefix for a in-preview command.
+      - There is not sub-grouping command for all experimental commands.
+    - A `warning note` should be displayed then the command is run with a note like:
+      > You are running an experimental command. It can be changed or removed in the future.
+
+1. Flags for command
+    - Follows the same rules from `1)`
+
+1. Field for `azure.yaml`
+    - There won't be changes to the `yaml-schema`. This means, setting fields for in-preview features will be syntax highlighted as an error. Customer should expect and ignore those errors.
+    - Similar to 1) and 2), when experimental fields are set within azure.yaml, there will be a warning message displayed while azd loads the `azure.yaml`.
+
```

Comment:
I think you're missing a group for "All other application features (providers, functionality, behavior)". I also think the general practice (aside from commands), is to do the following:

All features (aside from commands)
1. User should see a message like "Running with experimental feature '<feature>' enabled."